packages/sirius-web/backend/sirius-web-services/src/main/java/org/eclipse/sirius/web/services/documents/UploadDocumentEventHandler.java

```diff
@@ -223,46 +218,27 @@ private boolean containsProxies(Resource resource) {
     }
 
     /**
-     * Returns the {@link Resource} with the given {@link URI} or {@link Optional#empty()} regarding to the content of
-     * the first line of the given {@link InputStream}.
-     *
-     * <p>
-     * Returns a {@link JsonResourceImpl} if the first line contains a '{', a {@link XMIResourceImpl} if the first line
-     * contains '<', {@link Optional#empty()} otherwise.
-     * </p>
+     * Returns the {@link Resource} with the given {@link URI} or {@link Optional#empty()}.
      *
      * @param inputStream
      *            The {@link InputStream} used to determine which {@link Resource} to create
      * @param resourceURI
      *            The {@link URI} to use to create the {@link Resource}
      * @param resourceSet
      *            The {@link ResourceSet} used to store the loaded resource
-     * @return a {@link JsonResourceImpl}, a {@link XMIResourceImpl} or {@link Optional#empty()}
+     * @return a {@link Resource} or {@link Optional#empty()}
      */
     private Optional<Resource> getResource(InputStream inputStream, URI resourceURI, ResourceSet resourceSet) {
-        Resource resource = null;
-        BufferedInputStream bufferedInputStream = new BufferedInputStream(inputStream);
-        bufferedInputStream.mark(Integer.MAX_VALUE);
-        try (var reader = new BufferedReader(new InputStreamReader(bufferedInputStream, StandardCharsets.UTF_8))) {
-            String line = reader.readLine();
-            Map<String, Object> options = new HashMap<>();
-            if (line != null) {
-                if (line.contains("{")) {
-                    resource = new JSONResourceFactory().createResource(resourceURI);
-                } else if (line.contains("<")) {
-                    resource = new XMIResourceImpl(resourceURI);
-                    options = new EMFResourceUtils().getXMILoadOptions();
-                }
-            }
-            bufferedInputStream.reset();
-            if (resource != null) {
-                resourceSet.getResources().add(resource);
-                resource.load(bufferedInputStream, options);
-            }
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        try {
+            inputStream.transferTo(baos);
```

Comment:
Why do you need to copy everything? Why can't the loader methods take the inputStream as an input? This step was not necessary before.

Reply:
Because, before the change, the input was consumed and reset in the same method, and only once.
With the new `IResourceLoaderService`, the input can be consumed several times, and we don't want each implementation of `IResourceLoaderService` responsible for resetting the input.